jena-text/src/main/java/org/apache/jena/query/text/assembler/TextIndexESAssembler.java

```diff
@@ -0,0 +1,129 @@
+
+    private static Logger LOGGER      = LoggerFactory.getLogger(TextIndexESAssembler.class) ;
+
+    protected static final String COMMA = ",";
+    protected static final String COLON = ":";
+    /*
+    <#index> a :TextIndexES ;
+        text:serverList "127.0.0.1:9300,127.0.0.2:9400,127.0.0.3:9500" ; #Comma separated list of hosts:ports
+        text:clusterName "elasticsearch"
+        text:shards "1"
+        text:replicas "1"
+        text:entityMap <#endMap> ;
+        .
+    */
+    
+    @SuppressWarnings("resource")
+    @Override
+    public TextIndex open(Assembler a, Resource root, Mode mode) {
+        try {
+            String listOfHostsAndPorts = GraphUtils.getAsStringValue(root, pServerList) ;
+            if(listOfHostsAndPorts == null || listOfHostsAndPorts.isEmpty()) {
+                throw new TextIndexException("Mandatory property text:serverList (containing the comma-separated list of host:port) property is not specified. " +
+                        "An example value for the property: 127.0.0.1:9300");
+            }
+            String[] hosts = listOfHostsAndPorts.split(COMMA);
+            Map<String,Integer> hostAndPortMapping = new HashMap<>();
+            for(String host : hosts) {
+                String[] hostAndPort = host.split(COLON);
+                if(hostAndPort.length < 2) {
+                    LOGGER.error("Either the host or the port value is missing.Please specify the property in host:port format. " +
+                            "Both parts are mandatory. Ignoring this value. Moving to the next one.");
+                    continue;
+                }
+                hostAndPortMapping.put(hostAndPort[0], Integer.valueOf(hostAndPort[1]));
+            }
+
+            String clusterName = GraphUtils.getAsStringValue(root, pClusterName);
+            if(clusterName == null || clusterName.isEmpty()) {
+                LOGGER.warn("ClusterName property is not specified. Defaulting to 'elasticsearch'");
+                clusterName = "elasticsearch";
+            }
+
+            String numberOfShards = GraphUtils.getAsStringValue(root, pShards);
+            if(numberOfShards == null || numberOfShards.isEmpty()) {
+                LOGGER.warn("shards property is not specified. Defaulting to '1'");
+                numberOfShards = "1";
+            }
+
+            String replicationFactor = GraphUtils.getAsStringValue(root, pReplicas);
+            if(replicationFactor == null || replicationFactor.isEmpty()) {
+                LOGGER.warn("replicas property is not specified. Defaulting to '1'");
+                replicationFactor = "1";
+            }
+
+            String indexName = GraphUtils.getAsStringValue(root, pIndexName);
+            if(indexName == null || indexName.isEmpty()) {
+                LOGGER.warn("index Name property is not specified. Defaulting to 'jena-text'");
+                indexName = "jena-text";
+            }
+
+            boolean isMultilingualSupport = false;
```

Comment:
The multilingual parameter is no longer used in the ES implementation, so this block of code can be removed.